lametro/events.py

```diff
@@ -1,10 +1,31 @@
 import datetime
+import logging
 
 from legistar.events import LegistarAPIEventScraper
 from pupa.scrape import Event, Scraper
 from legistar.base import LegistarScraper
 
 
+LOGGER = logging.getLogger(__name__)
+
+class UnmatchedEventError(Exception):
+    def __init__(self, events):
+        if type(events) is dictionary:
```

Comment:
one way to make this code more DRY is to define your message format at the top of your init method, then `.format` it with the events as needed.

```python
message_fmt = "Can't find companion for Event {0} at {1} on {2} - {3}"
# if dict
    message = message_fmt.format(...)
# if list
    message = ''
    for e in events:
       message +=  message_fmt.format(...)
```

Reply:
done! thanks for the tip.